File: app/src/main/java/com/parabola/newtone/mvp/view/PlaylistView.java
package com.parabola.newtone.mvp.view;

import com.parabola.domain.model.Track;
import com.parabola.domain.settings.ViewSettingsInteractor.TrackItemView;

import java.util.List;

import moxy.MvpView;
import moxy.viewstate.strategy.alias.AddToEndSingle;

@AddToEndSingle
public interface PlaylistView extends MvpView {

    void setPlaylistChangerActivation(boolean activate);

    void setTracksCount(int playlistSize);
    void setPlaylistTitle(String playlistTitle);
    void refreshTracks(List<Track> tracks);
    void setItemViewSettings(TrackItemView viewSettings);
    void setItemDividerShowing(boolean showed);

    void setCurrentTrack(int trackId);

}

File: app/src/main/java/com/parabola/newtone/presentation/playlist/playlist/PlaylistView.kt
package com.parabola.newtone.presentation.playlist.playlist

import com.parabola.domain.model.Track
import com.parabola.domain.settings.ViewSettingsInteractor.TrackItemView
import moxy.MvpView
import moxy.viewstate.strategy.alias.AddToEndSingle

@AddToEndSingle
interface PlaylistView : MvpView {
    fun setPlaylistTitle(playlistTitle: String)

    fun setPlaylistChangerActivation(activate: Boolean)

    fun refreshTracks(tracks: List<Track>)
    fun setTracksCount(playlistSize: Int)
    fun setCurrentTrack(trackId: Int)

    fun setItemViewSettings(viewSettings: TrackItemView)
    fun setItemDividerShowing(showed: Boolean)
}
